src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAIResponseChatClient.cs

```diff
@@ -0,0 +1,582 @@
+                                {
+                                    // If the type can't be serialized, skip it.
+                                }
+                            }
+
+                            yield return ResponseItem.CreateFunctionCallOutputItem(resultContent.CallId, result ?? string.Empty);
+                            break;
+                    }
+                }
+
+                continue;
+            }
+
+            if (input.Role == ChatRole.Assistant)
+            {
+                foreach (AIContent item in input.Contents)
+                {
+                    switch (item)
+                    {
+                        case TextContent textContent:
+                            yield return ResponseItem.CreateAssistantMessageItem(
+                                "msg_ignored",
+                                textContent.Text);
+                            break;
+
+                        case FunctionCallContent callContent:
+                            yield return ResponseItem.CreateFunctionCall(
+                                "msg_ignored",
+                                callContent.CallId,
+                                callContent.Name,
+                                BinaryData.FromBytes(JsonSerializer.SerializeToUtf8Bytes(
+                                    callContent.Arguments,
+                                    options.GetTypeInfo(typeof(IDictionary<string, object?>)))));
+                            break;
+                    }
+                }
+
+                continue;
+            }
+        }
+    }
+
+    /// <summary>Extract usage details from an <see cref="OpenAIResponse"/>.</summary>
+    private static UsageDetails? ToUsageDetails(OpenAIResponse? openAIResponse)
+    {
+        UsageDetails? ud = null;
+        if (openAIResponse?.Usage is { } usage)
+        {
+            ud = new()
+            {
+                InputTokenCount = usage.InputTokenCount,
+                OutputTokenCount = usage.OutputTokenCount,
+                TotalTokenCount = usage.TotalTokenCount,
+            };
+
+            if (usage.OutputTokenDetails is { } outputDetails)
+            {
+                ud.AdditionalCounts ??= [];
+
+                const string OutputDetails = nameof(usage.OutputTokenDetails);
+                ud.AdditionalCounts.Add($"{OutputDetails}.{nameof(outputDetails.ReasoningTokenCount)}", outputDetails.ReasoningTokenCount);
+            }
+        }
+
+        return ud;
+    }
+
+    /// <summary>Convert a sequence of <see cref="ResponseContentPart"/>s to a list of <see cref="AIContent"/>.</summary>
+    private static List<AIContent> ToAIContents(IEnumerable<ResponseContentPart> contents)
+    {
+        List<AIContent> results = [];
+
+        foreach (ResponseContentPart part in contents)
+        {
+            if (part.Kind == ResponseContentPartKind.OutputText)
+            {
+                results.Add(new TextContent(part.Text));
+            }
```

Comment:
Is the API inherently text-only right now, or are multimodal cases handled differently?